src/server.cpp

```diff
@@ -1500,18 +1500,20 @@ void Server::SendMovement(u16 peer_id)
 	Send(&pkt);
 }
 
-void Server::SendPlayerHPOrDie(PlayerSAO *playersao)
+void Server::SendPlayerHPOrDie(PlayerSAO *playersao, s16 oldhp)
 {
 	if (!g_settings->getBool("enable_damage"))
 		return;
 
 	u16 peer_id   = playersao->getPeerID();
 	bool is_alive = playersao->getHP() > 0;
 
-	if (is_alive)
-		SendPlayerHP(peer_id);
-	else
+	// Make sure that the player only dies once.
+	if (playersao->getHP() < oldhp && !is_alive) {
```

Comment:
This is silly, oldhp shouldn't need be passed as a parameter.   It seems to me that SendPlayerHPOrDie is an ill-conceived function, and should actually be Server::DamagePlayer(PlayerSAO *playersao, s16 damage).

The current fix you have here is a kludge, and a totally unnecessary one at that.